### ISSUE TITLE
vpn mode not work in android 4.4.2

### ISSUE THREAD

**Issue by alirezaimi** [guardianproject/orbot#192]
Hi
when i run vpn mode all connection gone and i can not surf the net any more ! 
Why ?!
orbot 16.0.5-RC-2-tor-0.3.4.9
android 4.4.2

**Comment by A-Shahbazi** [comment]
I can also experience this on android 4.4.4
Here's the logcat immediately after enabling vpn mode:
```
I/Timeline( 2632): Timeline: Activity_launch_request id:org.torproject.android time:13577464
I/ActivityManager(  667): START u0 {cmp=org.torproject.android/.vpn.VPNEnableActivity} from pid 2632
D/VPNEnableActivity( 2632): prompting user to start Orbot VPN
W/VPNEnableActivity( 2632): prompt for VPN
I/Timeline( 2632): Timeline: Activity_launch_request id:com.android.vpndialogs time:13577559
I/ActivityManager(  667): START u0 {cmp=com.android.vpndialogs/.ConfirmDialog} from pid 2632
I/Timeline( 2632): Timeline: Activity_idle id: android.os.BinderProxy@41f37598 time:13577849
I/ActivityManager(  667): Displayed com.android.vpndialogs/.ConfirmDialog: +467ms (total +626ms)
I/Timeline( 4797): Timeline: Activity_idle id: android.os.BinderProxy@41b7c420 time:13578162
I/Timeline(  667): Timeline: Activity_windows_visible id: ActivityRecord{4212c128 u0 org.torproject.android/.vpn.VPNEnableActivity t9} time:13578324
I/Timeline(  667): Timeline: Activity_windows_visible id: ActivityRecord{4249f7f0 u0 com.android.vpndialogs/.ConfirmDialog t9} time:13578325
W/AudioTrack(  667): AUDIO_OUTPUT_FLAG_FAST denied by client due to mismatching sample rate (48000 vs 44100)
D/AudioFlinger(  116): start(4097), calling pid 667 session 242
D/AudioFlinger(  116): ? => ACTIVE (4097) on thread 0x40bcc8f8
I/AudioHardwareMSM76XXA(  116): Routing audio to Speakerphone
I/Vpn     (  667): Switched from [Legacy VPN] to org.torproject.android
D/Vpn     (  667): setting state=IDLE, reason=prepare
D/OrbotVpnService( 2632): starting OrbotVPNService service!
I/Timeline( 2632): Timeline: Activity_idle id: android.os.BinderProxy@41f37598 time:13581015
I/OrbotVpnService( 2632): PDNSD: 0
W/System.err( 2632): java.net.BindException: bind failed: EADDRINUSE (Address already in use)
W/System.err( 2632): 	at libcore.io.IoBridge.bind(IoBridge.java:89)
W/System.err( 2632): 	at java.net.PlainSocketImpl.bind(PlainSocketImpl.java:150)
W/System.err( 2632): 	at java.net.ServerSocket.<init>(ServerSocket.java:101)
W/System.err( 2632): 	at com.runjva.sourceforge.jsocks.protocol.ProxyServer.start(ProxyServer.java:180)
W/System.err( 2632): 	at org.torproject.android.service.vpn.OrbotVpnManager$1.run(OrbotVpnManager.java:182)
W/System.err( 2632): Caused by: libcore.io.ErrnoException: bind failed: EADDRINUSE (Address already in use)
W/System.err( 2632): 	at libcore.io.Posix.bind(Native Method)
D/Vpn     (  667): setting state=CONNECTING, reason=establish
W/System.err( 2632): 	at libcore.io.ForwardingOs.bind(ForwardingOs.java:40)
W/System.err( 2632): 	at libcore.io.IoBridge.bind(IoBridge.java:87)
D/VpnJni  (  667): Address added on tun0: 192.168.200.1/32
W/System.err( 2632): 	... 4 more
I/Vpn     (  667): Established by org.torproject.android on tun0
D/Vpn     (  667): setting state=AUTHENTICATING, reason=establish
D/Tun2Socks( 2632): NOTICE(tun2socks): initializing BadVPN tun2socks 1.999.129
D/Tun2Socks( 2632): NOTICE(tun2socks): entering event loop
I/Timeline( 2632): Timeline: Activity_idle id: android.os.BinderProxy@41b73f30 time:13582305
I/Timeline(  667): Timeline: Activity_windows_visible id: ActivityRecord{420df838 u0 org.torproject.android/.OrbotMainActivity t9} time:13582494
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 42 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 70 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: from device 1440 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: from device 226 bytes
D/Orbot   ( 2632): orConnStatus (NX01): CLOSED
D/Orbot   ( 2632): Circuit (367) CLOSED: NX01 > pulsetor > marcuse1
D/Orbot   ( 2632): Circuit (363) CLOSED: NX01 > arbitraryKenzie1 > IPredator
D/Orbot   ( 2632): Circuit (368) CLOSED: NX01 > FreewaySca > IPredator
D/Orbot   ( 2632): Circuit (364) CLOSED: NX01 > swagspacetor1 > NetnikTorExit1
D/Orbot   ( 2632): Circuit (362) CLOSED: NX01 > osipetz > sofia
D/Orbot   ( 2632): Circuit (369) CLOSED: NX01 > TotorBE1 > niftychionomy
D/Tun2Socks( 2632): INFO(tun2socks): UDP: from DNS 42 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 70 bytes
D/Orbot   ( 2632): Circuit (366) CLOSED: NX01 > paprika > Urania
D/Orbot   ( 2632): Circuit (365) CLOSED: NX01 > geekbox > DigiGesTor5e1
D/Orbot   ( 2632): Circuit (370) LAUNCHED: 
D/Orbot   ( 2632): orConnStatus (NX01): LAUNCHED
D/Tun2Socks( 2632): INFO(tun2socks): UDP: from device 60 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: from device 40 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: from device 1500 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: from device 1500 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: from device 442 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 42 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 70 bytes
D/Orbot   ( 2632): Circuit (371) LAUNCHED: 
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Orbot   ( 2632): Circuit (372) LAUNCHED: 
D/Orbot   ( 2632): Circuit (373) LAUNCHED: 
D/Orbot   ( 2632): Circuit (374) LAUNCHED: 
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 42 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 70 bytes
D/Orbot   ( 2632): Circuit (375) LAUNCHED: 
D/Orbot   ( 2632): Circuit (376) LAUNCHED: 
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 42 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 70 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 42 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 70 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 42 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 70 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 42 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 70 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 42 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 70 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 42 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 70 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 42 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 70 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 42 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 70 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 42 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 70 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 42 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 70 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 42 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 70 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: to DNS 37 bytes
D/Tun2Socks( 2632): INFO(tun2socks): UDP: sending modified DNS packet out 65 bytes
```

**Comment by abdolfazel** [comment]
I have the same problem in Android 4.4.2 but I think Iran government filtered (blocked) VPN so we can't use VPN but I think Iran decrease internet speed when you turn on VPN

**Comment by A-Shahbazi** [comment]
> I have the same problem in Android 4.4.2 but I think Iran government filtered (blocked) VPN so we can't use VPN but I think Iran decrease internet speed when you turn on VPN

Obviously irrelevant.
This is an issue with vpn implementation for android KitKat. Apps that are able to use socks5 proxies do work well.
BTW, I have as well tested on another phone with android 8 and vpn works well.

**Comment by n8fr8** [comment]
We've released Orbot 16.3.1 with many important fixes. please test!
